src/Symfony/Component/Notifier/Bridge/LightSms/LightSmsTransport.php

```diff
@@ -0,0 +1,171 @@
+        '29' => 'Missing end date (Format: YYYY-MM-DD)',
+        '30' => 'No date (format: YYYY-MM-DD)',
+        '31' => 'Closing direction to the user',
+        '32' => 'Not enough money',
+        '33' => 'Missing phone number',
+        '34' => 'Phone is in stop list',
+        '35' => 'Not enough money',
+        '36' => 'Can not obtain information about phone',
+        '37' => 'Base Id is not set',
+        '38' => 'Phone number already exists in this database',
+        '39' => 'Phone number does not exist in this database',
+    ];
+
+    public function __construct(string $login, string $password, string $phone, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->login = $login;
+        $this->password = $password;
+        $this->phone = $phone;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('lightsms://%s?phone=%s', $this->getEndpoint(), $this->phone);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $timestamp = time();
+
+        $signature = $this->generateSignature([
+            'message' => $message,
+            'timestamp' => $timestamp,
+        ]);
+
+        $endpoint = sprintf(
+            'https://%s/external/get/send.php?login=%s&signature=%s&phone=%s&text=%s&sender=%s&timestamp=%s',
+            $this->getEndpoint(),
+            $this->login,
+            $signature,
+            $this->escapePhoneNumber($message->getPhone()),
+            $this->escapeSubject($message->getSubject()),
+            $this->phone,
+            $timestamp
+        );
+
+        $response = $this->client->request('GET', $endpoint);
+
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
+            throw new TransportException('Unable to send the SMS: ', $response);
+        }
+
+        $content = $response->toArray(false);
+
+        if (isset($content['error'])) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$content['error']], $response);
+        }
+
+        $phone = $this->escapePhoneNumber($message->getPhone());
+        if (32 === $content[$phone]['error']) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$content['error']], $response);
+        }
+
+        if (0 == $content[$phone]['error']) {
+            $sentMessage = new SentMessage($message, (string) $this);
+            $sentMessage->setMessageId($content[$phone]['id_sms']);
+        }
+
+        return $sentMessage;
+    }
+
+    private function generateSignature(array $params): string
```

Comment:
Why the `array` and not:
```suggestion
    private function generateSignature(SmsMessage $message, string $timestamp): string
```
?

Reply:
Fixed.